engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/CallArgument.scala

```diff
@@ -138,7 +138,7 @@ object CallArgument {
          |""".toSingleLine
 
     /** @inheritdoc */
-    override def children: List[IR] = name.toList :+ value
+    override lazy val children: List[IR] = name.toList :+ value
```

Comment:
Not related to the change, but it should've been Vector, instead of a List

Reply:
Once I heard @4e6 saying we shouldn't use `List` as it does an _eager copy_, but a lazy iterator. With `List` the `preorder()` implementation:
- calls `children` on each one
- the `children` implementation constructs a `List`
- the `preoder` then concatenates the list to its own `List` representing the whole `IR` tree

Where can the **slowdown** be? Only in the construction of of `children` `List`, right? This PR tries to trade that CPU cost for increased memory consumption.

Wouldn't it be better to change the `children` to follow more efficient pattern and not allocate its own `List`?

Reply:
> elements https://august.nagro.us/list-vs-vector.html

Our usecase is "Construction with Append + Traversals" where Vector is faster. Possibly even better with `children(addTo : Vector.Builder) : Unit`.

Up to you guys, I don't want to be expert on Scala data structures.

Reply:
> I don't want to be expert on Scala data structures.

I continue to believe the best might be to have a _lazy iterator_ like in this [case of iterating all children of a directory](https://github.com/apache/netbeans/blob/cc6815804038b9d8a45171eb34131a5d51a04d07/platform/openide.filesystems/src/org/openide/filesystems/FileObject.java#L918) - e.g. a _queue_ where each `nextElement()` adds its children to the end of the queue.